common/lib/dependabot/pull_request_creator/branch_namer/dependency_group_strategy.rb

```diff
@@ -37,11 +37,20 @@ def prefixes
           ].compact
         end
 
-        # When superseding a grouped update pull request, we will have a period
-        # of time when there are two branches for the group so we use a timestamp
-        # to avoid collisions.
-        def timestamped_group_name
-          "#{dependency_group.name}-#{Time.now.utc.to_i}"
+        # Group pull requests will generally include too many dependencies to include
+        # in the branch name, but we rely on branch names being deterministic for a
+        # given set of dependency changes.
+        #
+        # Let's append a short hash digest of the dependency changes so that we can
+        # meet this guarantee.
+        def group_name_with_dependency_digest
+          "#{dependency_group.name}-#{dependency_digest}"
+        end
+
+        def dependency_digest
+          @dependency_digest ||= Digest::MD5.hexdigest(dependencies.map do |dependency|
```

Comment:
Is the order of the dependencies deterministic? If not the hash will be different each run.

Reply:
Fixed in https://github.com/dependabot/dependabot-core/pull/7365/commits/ae54458d5e371ffc9a3c5a187271c095b05c1b52